src/main/java/org/ltr4l/nn/AbstractNode.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018 org.LTR4L
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ltr4l.nn;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Node in the network.
+ * Holds information regarding the edges (which nodes are connected), and based on that information
+ * the total input and output. Also contains information about Activation.
+ */
+public class AbstractNode<E extends AbstractEdge> {
```

Comment:
Why isn't this abstract class? Though the name of the class has "abstract" but the class is not abstract seems strange to me.

Reply:
Apologies, I meant for it to be abstract. I will change this too.

Reply:
Will keep it a regular class and change the name.